### PR TITLE
Implement tj command.

### PR DESCRIPTION
**Pull request by giuscri**
See #9073.

Example:

```
[0xf7fccc60]> t?~json
| tj                      List all loaded types as json
[0xf7fccc60]> tj
[{"type":"char","size":8,"format":"c"},{"type":"char *","size":32,"format":"z"},{"type":"int","size":32,"format":"d"},{"type":"int16_t","size":16,"format":"w"},{"type":"int32_t","size":32,"format":"d"},{"type":"int64_t","size":64,"format":"q"},{"type":"int8_t","size":8,"format":"b"},{"type":"long","size":64,"format":"x"},{"type":"long long","size":64,"format":"q"},{"type":"short","size":16,"format":"w"},{"type":"size_t","size":32,"format":"d"},{"type":"uid_t","size":64,"format":"i"},{"type":"uint16_t","size":16,"format":"w"},{"type":"uint32_t","size":32,"format":"d"},{"type":"uint64_t","size":64,"format":"q"},{"type":"uint8_t","size":8,"format":"b"},{"type":"unsigned char","size":8,"format":"b"},{"type":"unsigned int","size":32,"format":"i"},{"type":"unsigned short","size":16,"format":"w"},{"type":"void *","size":32,"format":"p"}]
```

### CODE REVIEW

File: libr/core/cmd_type.c [hunk 1]
```diff
@@ -311,31 +312,57 @@ static int typelist(void *p, const char *k, const char *v) {
...
+
+static int sdbforcb_json (void *p, const char *k, const char *v) {
+	Sdb *sdb = (Sdb *)p;
+	char *sizecmd = malloc (strlen("type..size") + strlen(k) + 1);
+	char *formatcmd = malloc (strlen("type.") + strlen(k) + 1);
+	sprintf (sizecmd, "type.%s.size", k);
+	sprintf (formatcmd, "type.%s", k);
+	r_cons_printf ("{\"type\":\"%s\",\"size\":%d,\"format\":\"%c\"}", k,
+			atoi(sdb_querys(sdb, NULL, -1, sizecmd)),
+			sdb_querys(sdb, NULL, -1, formatcmd)[0]);
```
Comment (giuscri) [inline libr/core/cmd_type.c]: Are all formats 1-character only? Is that a valid assumption?

File: libr/core/cmd_type.c [hunk 2]
```diff
@@ -311,31 +312,57 @@ static int typelist(void *p, const char *k, const char *v) {
...
-		r_cons_println (k);
-	}
+static int sdbforcb_default (void *p, const char *k, const char *v) {
+	r_cons_println (k);
+	return 1;
+}
+
+static int sdbforcb_json (void *p, const char *k, const char *v) {
+	Sdb *sdb = (Sdb *)p;
+	char *sizecmd = malloc (strlen("type..size") + strlen(k) + 1);
```
Comment (radare) [inline libr/core/cmd_type.c]: wtf? use r_str_newf

File: libr/core/cmd_type.c [hunk 3]
```diff
@@ -311,31 +312,57 @@ static int typelist(void *p, const char *k, const char *v) {
...
+	char *formatcmd = malloc (strlen("type.") + strlen(k) + 1);
+	sprintf (sizecmd, "type.%s.size", k);
+	sprintf (formatcmd, "type.%s", k);
+	r_cons_printf ("{\"type\":\"%s\",\"size\":%d,\"format\":\"%c\"}", k,
+			atoi(sdb_querys(sdb, NULL, -1, sizecmd)),
```
Comment (radare) [inline libr/core/cmd_type.c]: missing spaces before every (

File: libr/core/cmd_type.c [hunk 4]
```diff
@@ -311,31 +312,57 @@ static int typelist(void *p, const char *k, const char *v) {
...
-			sdbforcb ((void *)core, kv->key, kv->value);
+			if (!strncmp (kv->value, "type", strlen ("type") + 1)) {
+				ls_append (filtls, kv);
+			}
+		}
+		if (mode == 'j') {
+			r_cons_print ("[");
+			ls_foreach (filtls, it, kv) {
+				sdbforcb_json ((void *)core->anal->sdb_types, kv->key, kv->value);
+				if (it->n) r_cons_print (",");
```
Comment (radare) [inline libr/core/cmd_type.c]: missing braces

File: libr/core/cmd_type.c (hunk 1 above)
Comment (radare) [reply libr/core/cmd_type.c]: dont think so, a char* should be valid

Comment (giuscri) [reply libr/core/cmd_type.c]: problem was sdb_querys returns a string ending with `\n`. I don't know how to trim it — without writing too much.

Comment (giuscri) [reply libr/core/cmd_type.c]: oh ...r_str_trim.

**Comment by giuscri** [comment]
OK @radare, I should have fixed that

**Pull request merged**